### PR TITLE
Playbook, Workbook & Analytic Rule for the Armorblox Sentinel integration

### PR DESCRIPTION
**Pull request by urajat**
Update Solutions/Armorblox/Analytic Rules/ArmorbloxNeedsReviewAlert.yaml
Update Solutions/Armorblox/Playbooks/README.md

Co-authored-by: Rajat Upadhyaya <45485+urajat@users.noreply.github.com>

## Before submitting this PR please ensure that you have read the following sections and then completed the template below:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures, there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Added Playbook, Workbook and an Analytic Rule for the Armorblox Sentinel integration

## After the submission has been made, please look at the Validation Checks:

> Check that the validations are passing and address any issues that are present. Let us know if you have tried fixing and need help.

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

## PR Template

-----------------------------------------------------------------------------------------------------------
 **Description for the PR:**

Added Playbook, Workbook and an Analytic Rule for the Armorblox Sentinel integration

**Testing Completed:**
  Yes/ No : Yes

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by ankita-sharma2019** [comment]
Hello @aprakash13, @v-jayakal ,

By when could we expect your review/feedback on this PR? Is there something missing on our end to be added?

Thanks

**Review by sreedharande — DISMISSED** [review]

**Comment by sreedharande** [comment]
Reviewed Playbook

**Comment by v-sabiraj** [comment]
Hey @urajat , workbook queries are failing , please update , thanks.
image

**Comment by v-sabiraj** [comment]
image
Hey @urajat as **[UTC]** is added for **date_t** in sample data and in query just  **date_t** used, because of that it may give query error, please update , thanks.

**Comment by urajat** [comment]
@v-sabiraj Thanks for the review. The 2 issues have been fixed.

**Comment by v-sabiraj** [comment]
Hey @urajat , as 'id' has some numeric value , the id_s should be replaced with id_d , please check and update , thanks.

**Comment by v-sabiraj** [comment]
Hey @urajat ,the workbook query for "date_t" is still failing, please check and update, thanks.
image

**Comment by v-sabiraj** [comment]
> Hey @urajat ,the workbook query for "date_t" is still failing, please check and update, thanks. image

Hey @urajat / @xoriant-prateekg ,just please check with date_t , it must also be date_s, please check and update thanks.

**Comment by v-sabiraj** [comment]
@aprakash13 ,please review Analytic Rules,thanks.

**Comment by v-sabiraj** [comment]
> > Hey @urajat ,the workbook query for "date_t" is still failing, please check and update, thanks. image
> 
> Hey @urajat / @xoriant-prateekg ,just please check with date_t , it must also be date_s, please check and update thanks.

@urajat, @xoriant-prateekg ,please update the requested changes, thanks.

**Comment by v-sabiraj** [comment]
@urajat ,please change date_t to date_s in sample data, thanks.

**Comment by urajat** [comment]
@v-sabiraj Thanks, we were wondering if changing `date_t` to `date_s` would mess up existing installations of the Data Connector?

**Comment by xoriant-prateekg** [comment]
> Hey @urajat , please refer the [template](https://docs.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#:~:text=KQL%20provides%20the%20datetime%20data%20type%20for%20date,YYYY-MM-DD%20YYYY-MM-DDThh%3Amm%3Ass%20YYYY-MM-DDThh%3Amm%3AssZ%20YYYY-MM-DDThh%3Amm%3AssfrZ%20In%20these%20datetime%20formats%3A) for the correct format of date_t or change to date_s, thanks.

Updated the sample data JSON. Changed the format of timestring to ISO-8601 compatible format.

**Comment by v-sabiraj** [comment]
Hey @aprakash13 , please review and approve Analytic Rules, thanks.

**Comment by v-sabiraj** [comment]
Workbook looks good, @aprakash13 ,please review Analytic Rules, thanks.

**Comment by v-sabiraj** [comment]
Hey @urajat ,please fix the license/cla issue, thanks.

**Comment by xoriant-prateekg** [comment]
Hi @v-sabiraj, I signed the CLA but the license/cla check is still not reflecting the same. Should we wait for the confirmation or is there any other action item for us as?
Screenshot from 2022-02-28 17-15-44

**Comment by v-sabiraj** [comment]
Hey @xoriant-prateekg ,please refer this pr ([licence/cla — Waiting for status to be reported (https://github.com/PBSPro/pbspro/pull/68)](https://github.com/cla-assistant/cla-assistant/issues/124#)
#124) for your reference, thanks.

**Comment by v-sabiraj** [comment]
Hey @xoriant-prateekg ,You can "recheck" the PR of your repository by pushing the "Recheck PRs" button in CLA assistant, thanks.

**Comment by v-sabiraj** [comment]
Hey @aprakash13 , please review and approve Analytic Rules, thanks.

**Comment by v-sabiraj** [comment]
Hey @aprakash13 , please review and approve Analytic Rules, thanks.

**Comment by v-sabiraj** [comment]
Hey @aprakash13 , please review and approve Analytic Rules, thanks.

**Comment by v-sabiraj** [comment]
Hey @aprakash13 , will you please review and approve Analytic Rules, thanks.

File: Solutions/Armorblox/Analytic Rules/ArmorbloxNeedsReviewAlert.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Armorblox
+    dataTypes:
+      - Armorblox_CL
+queryFrequency: 10m
+queryPeriod: 10m
+triggerOperator: GreaterThan
+triggerThreshold: 0
+query: "Armorblox_CL\r\n| where remediation_actions_s contains \"Needs Review\""
```
Comment (aprakash13) [inline Solutions/Armorblox/Analytic Rules/ArmorbloxNeedsReviewAlert.yaml:14]: Please remove the quotes at the start and the end. This is not needed. Also remove the \r\n that probably is not needed.
The \ also seems to be not required. Would you please remove that as well? Thanks

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Thanks for submitting this query.
Please look and address the small suggestions being made.

**Comment by v-sabiraj** [comment]
Hey @urajat/ @xoriant-prateekg , can you please resolve the requested changes, thanks.

File: Solutions/Armorblox/Analytic Rules/ArmorbloxNeedsReviewAlert.yaml (hunk 1 above)
Comment (urajat) [reply Solutions/Armorblox/Analytic Rules/ArmorbloxNeedsReviewAlert.yaml]: This has been done

**Review by aprakash13 — APPROVED** [review]
Approving this from a query perspective.

**Pull request merged**